04-onion-routing.md

```diff
@@ -755,59 +764,162 @@ Note that these error messages are not reliable, as they are not placed on-chain
 due to the possibility of hop failure.
 
 Intermediate hops store the shared secret from the forward path and reuse it to
-obfuscate any corresponding return packet during each hop.
+authenticate and obfuscate any corresponding return packet during each hop.
 In addition, each node locally stores data regarding its own sending peer in the
 route, so it knows where to return-forward any eventual return packets.
-The node generating the error message (_erring node_) builds a return packet
+
+The origin node signals to each node on the path that it supports the error
+format that is detailed below by setting the `attributable_error` tlv record in
+the forward packet to contain the desired structure. It must make sure that each
+node supports this format by observing the node feature
+`option_attributable_error`.
+
+By choosing a structure for the attributable errors, the origin node is making a
+trade-off between failure message size and privacy. For maximum privacy,
+`hop_count` should be set to the global maximum of 27. A lower value reveals a
```

Comment:
Where does that maximum come from? It used to be 20 fixed, however we gained a bit of room by switching to TLV payloads, but I wasn't aware it is 7 more hops than initially specced.

Reply:
This is the lnd version of the calculation: https://github.com/lightningnetwork/lightning-onion/blob/ca23184850a16cd14d27619f3afdae543b3857a9/path.go#LL315C18-L315C18

The difference seems to be the _truncated_ values for amt and timelock.